src/type.zig

```diff
@@ -2643,7 +2643,7 @@ pub const Type = extern union {
             => ty.childType().hasWellDefinedLayout(),
 
             .optional => ty.isPtrLikeOptional(),
-            .@"struct" => ty.castTag(.@"struct").?.data.layout != .Auto,
+            .@"struct" => ty.castTag(.@"struct").?.data.layout == .Extern,
```

Comment:
I believe this should eventually look like the case for enums above, packed struct with an explicit backing integer type do have well defined memory layouts as per #5049.